### ISSUE TITLE
Why tooltip and popover prioritze element data instead of explicit options?

### ISSUE THREAD

**Issue by cervengoc** [twbs/bootstrap#18329]
Please explain what the reason is behind the behaviour of `Tooltip.prototype.getTitle` and `Popover.prototype.getContent` methods? They prioritize the info from the actual element instead of prioritizing the explicit  `options.title` and `options.content`.

This is an issue for me because I'm trying to create popovers dynamically and whenever an element has some `title` I have no choice to customize the popover content.

**Comment by cvrebert** [comment]
Duplicate of #8961.

**Comment by cvrebert** [comment]
Regardless of which behavior is chosen, the precedence of passed-in options vs. `data-*` attributes bears documenting. It doesn't seem to be documented currently.

**Comment by cervengoc** [comment]
@cvrebert Thanks for pointing out that this issue was already discussed earlier. I haven't find it.

I still would argue with this design decision. My current scenario is that I would like to customize the tooltip title to allow showing an optional abbrevation for an item. For this I created this setup:

``` 
$(document).popover({
    selector: ".help-icon",
    trigger: tipTriggers,
    delay: 500,
    container: 'body',
    html: true,
    placement: 'auto',
    title: elem => {
      var caption = $(elem).attr("title");
      var abbrevation = $(elem).attr("data-abbrevation");
      return caption && abbrevation && caption != abbrevation ? `<b>${abbrevation}</b><br/>${caption}` : caption;
    }
  });
```

This unexpectadly not working because Bootstrap insists to use the raw title attribute and never allows overriding.

So if not this way, then how could I achieve this without serious hacking of the library? Thanks for the help in advance.

**Comment by cvrebert** [comment]
CC: @fat

**Comment by mdo** [comment]
Seems to have been addressed with #19541.
